src/ChatGptNet/ChatGptServiceCollectionExtensions.cs

```diff
@@ -9,6 +9,52 @@ namespace ChatGptNet;
 /// </summary>
 public static class ChatGptServiceCollectionExtensions
 {
+    /// <summary>
+    /// Registers a <see cref="ChatGptClientFactory"/> instance with the specified options.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <param name="setupAction">The <see cref="Action{ChatGptOptions}"/> to configure the provided <see cref="ChatGptOptions"/>.</param>
+    /// <returns>A reference to this instance after the operation has completed.</returns>
+    public static IServiceCollection AddChatGptClientFactory(this IServiceCollection services, Action<ChatGptOptions> setupAction)
+    {
+        ArgumentNullException.ThrowIfNull(services);
+        ArgumentNullException.ThrowIfNull(setupAction);
+
+        var options = new ChatGptOptions();
+        setupAction.Invoke(options);
+
+        services.AddMemoryCache();
+        services.AddSingleton<IChatGptClientFactory, ChatGptClientFactory>(
```

Comment:
Why are you using this method and not simply, after adding `ChatGptOptions `itself to the service collection, like in https://github.com/marcominerva/ChatGptNet/blob/2193d2c7b9656d337cbd24b344b8ea3bba995fbb/src/ChatGptNet/ChatGptServiceCollectionExtensions.cs#L28):

`services.AddSingleton<IChatGptClientFactory, ChatGptClientFactory>();`